service/file_controller_create_volume.go

```diff
@@ -186,3 +206,68 @@ func queryShareData(conn RestEndpoint, fsName string) (map[string]string, error)
 	}
 	return nfsShareMetaData, err
 }
+
+func queryFileInterface(conn RestEndpoint, nasId string) (*list.List, error){
+	var getFileInterfaceUrl string = "/api/types/fileInterface/instances?fields=id,nasServer"
+    encodedQuery := EncodeUrl(getFileInterfaceUrl)
+    logrus.Debug("Encoded: ", encodedQuery)
+    status, resp := conn.get(encodedQuery)
+    logrus.Debug("Status: ", status)
+    logrus.Debug("Query file interface response: ", resp)
+    jsonParsed, _ := gabs.ParseJSON([]byte(resp))
+    entryCount := jsonParsed.Path("entryCount").Data().(float64)
+    fileInterfaces := list.New();
+
+
+    var err error = nil
+    if int(entryCount) == 0 {
+    	logrus.Info("Info: the num of file interface entries is 0")
+    	//err = errors.New(fmt.Sprintf("Error: no file interface created for nas server:%s ", nasId))
+	}else{
+		children, _ := jsonParsed.S("entries").Children()
+		for _, child := range children{
+			if child.Path("content").Path("nasServer").Data().(string) == nasId {
+				fileInterfaceData := make(map[string]string)
+				fileInterfaceData["id"] = child.Path("content").Path("id").Data().(string)
+				fileInterfaces.PushBack(fileInterfaceData);
+				logrus.Info(fileInterfaceData)
+			}
+		}
+		logrus.Info(fileInterfaces)
+	}
+    return fileInterfaces, err
+}
+
+func createFileInterface(conn RestEndpoint, fileIPInput fileInterfaceInput)(map[string]string, error) {
+	var err error = nil
+	fileInterfaceData := make(map[string]string)
+
+	fileInterfaces, _ := queryFileInterface(conn, fileIPInput.nasServer)
+	if fileInterfaces.Len() == 0 {
+		logrus.Info("Info: file interface is not created for nas server %s", fileIPInput.nasServer)
+		// start to create interface
+		var createFileInterfaceUrl string = "/api/types/fileInterface/instances"
+		// Body: {"nasServer": {"id":"nas_1"}, "ipPort": {"id":"spa_iom_0_eth0"}, "ipAddress": "10.103.76.143","netmask":"255.255.248.0", "gateway":"10.103.72.1"}
+		var createFileInterfaceBody string = fmt.Sprintf(`{"nasServer": {"id":"%s"}, "ipPort": {"id":"%s"}, "ipAddress":"%s", "netmask":"%s", "gateway":"%s"}`, fileIPInput.nasServer, fileIPInput.ipPort, fileIPInput.ipAddress, fileIPInput.netmask, fileIPInput.gateway)
+		status, resp := conn.post(createFileInterfaceUrl, createFileInterfaceBody)
+		logrus.Debug("Status: ", status)
+		logrus.Debug("Create file interface response: ", resp)
+		jsonParsed, _ := gabs.ParseJSON([]byte(resp))
+		content := jsonParsed.Path("content").Data().(float64)
+
+		if int(content) != 1 {
+			logrus.Error("Error: failed to create file interface for nas server %s", fileIPInput.nasServer)
+			err = errors.New(fmt.Sprintf("Error: failed to create file interface for nas server:%s", fileIPInput.nasServer))
+		}else{
+			children, _ :=jsonParsed.S("content").Children()
+			for _, child := range children{
+				fileInterfaceData["id"] =  child.Path("id").Data().(string)
```

Comment:
Forgot to return fileInterfaceData

Reply:
intended to forget, haha. add it in latest commit for future use. Thanks for review and comments.